app/services/discussion_service.rb

```diff
@@ -40,22 +40,14 @@ def self.update(discussion:, params:, actor:)
     end
 
     return false unless discussion.valid?
-
-    update_search_vector = discussion.title_changed? || discussion.description_changed?
-
-    event = true
-    if discussion.title_changed?
-      event = Events::DiscussionTitleEdited.publish!(discussion, actor)
-    end
-
-    if discussion.description_changed?
-      event = Events::DiscussionDescriptionEdited.publish!(discussion, actor)
-    end
-
+    return discussion if discussion.changed == ['uses_markdown']
+    return discussion unless discussion.changed?
+    
+    event = Events::DiscussionEdited.publish!(discussion, actor)
```

Comment:
This is nitpicky, but I'd rather save the discussion before we publish the event, just in case something goes wrong (event though we've already verified the discussion is valid)